app/scripts/lib/typed-message-manager.js

```diff
@@ -192,11 +192,13 @@ export default class TypedMessageManager extends EventEmitter {
           data.primaryType in data.types,
           `Primary type of "${data.primaryType}" has no type definition.`,
         );
-        assert.equal(
-          validation.errors.length,
-          0,
-          'Signing data must conform to EIP-712 schema. See https://git.io/fNtcx.',
-        );
+        if (validation.errors.length !== 0) {
+          throw new EthereumRpcError(
+            -32602,
```

Comment:
I hate to ask but should we create a constants file where we name RPC error codes with descriptive vars?  `-32602` is a magic number and most engineers wouldn't immediately know what that represents.